pkg/querier/queryrange/detected_fields.go

```diff
@@ -0,0 +1,370 @@
+package queryrange
+
+import (
+	"context"
+	"net/http"
+	"slices"
+	"strconv"
+	"time"
+
+	"github.com/axiomhq/hyperloglog"
+	"github.com/dustin/go-humanize"
+	"github.com/grafana/dskit/httpgrpc"
+	"github.com/prometheus/prometheus/model/labels"
+
+	"github.com/grafana/loki/v3/pkg/logproto"
+	logql_log "github.com/grafana/loki/v3/pkg/logql/log"
+	"github.com/grafana/loki/v3/pkg/logql/syntax"
+	"github.com/grafana/loki/v3/pkg/logqlmodel"
+	"github.com/grafana/loki/v3/pkg/querier/plan"
+	base "github.com/grafana/loki/v3/pkg/querier/queryrange/queryrangebase"
+
+	"github.com/grafana/loki/pkg/push"
+)
+
+func NewDetectedFieldsHandler(
+	limitedHandler base.Handler,
+	logHandler base.Handler,
+	limits Limits,
+) base.Middleware {
+	return base.MiddlewareFunc(func(next base.Handler) base.Handler {
+		return base.HandlerFunc(
+			func(ctx context.Context, req base.Request) (base.Response, error) {
+				r, ok := req.(*DetectedFieldsRequest)
+				if !ok {
+					return next.Do(ctx, req)
```

Comment:
should we return an error here instead? 
I might have missed something, but this seems to go to the nextRT eventually ending up in the querier which is not intended..

Reply:
not sure I follow about it ending up in the querier (we want it to end up there), but yes, we should be able to return an error here since this is only used in the detected fields round tripper (and wouldn't work anywhere else)